### ISSUE TITLE
Accessibility: Smart Autoannounce not working

### ISSUE THREAD

**Issue by fcnjd** [osmandapp/OsmAnd#19209]
### Description

Since I am blind, I was looking for the accessibility services. However, sadly I found that some of them, including the smart autoannounce, are not working.

### Steps to reproduce

1. Enable accessibility extension
2. Go to settings, walking profile, accessibility and turn accessibility mode on
3. Enable smart autoannounce and set the period to 10s
4. Start navigating a route

### Actual result

There's no smart autoannounce given, directions are only announced 20m-30m before, but even if I have to go several 100m streight I don't hear anything during that.

### Expected result

I expect smart autoannounce to work as I turn this setting on, announcing directions or route info approximately every 10s if this period is set.

### Your Environment (required)

```
OsmAnd Version: 4.5.10
Android/iOS version: Android 14
Device model: Google Pixel 7
```

**Comment by scaidermern** [comment]
Can confirm with OsmAnd 4.7.1.

**Comment by yuriiurshuliak** [comment]
The bug has been reproduced. Here are the steps to reproduce:

- Main menu > Plugins > Enable the Accessibility plugin
- Turn on accessibility mode on the Android system.
- Menu > Settings >Pedestrian profile settings > Accessibility > Enable smart autoannounce and set the period to 10s > Start navigation from point to point on a straight road without turns.

As a result, Autoannounce period does not provide any voice announcements if you follow a straight path.

OsmAnd~ 4.7.0#1726m, released: 2024-03-01

<table style="width:100%">
  </tr>
  <tr>
    <td> <video source src="https://github.com/osmandapp/OsmAnd/assets/127092082/71498df0-7585-44b5-b3a7-c1d88dbc763c"  width="300" height="auto"/>   </td>
     <th><img src="https://github.com/osmandapp/OsmAnd/assets/127092082/5e1f6e31-4b04-4e5e-93a2-400239159913"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by vshcherb** [comment]
I think now we need to double verify if it works correctly and document it. https://github.com/osmandapp/OsmAnd/commit/e4562bc89745fd17a03d6202e25ad8f4ce6bb33c

**Comment by vshcherb** [comment]
1. You need to enable smart notification once you click on my location
2. Notifications and checks are controlled by ACCESSIBILITY_AUTOANNOUNCE_PERIOD - by default 10 seconds. So you shouldn't hear any notification more frequent then this setting. 
3. Smart autoannounance
- Smart autoannounance on - in case target point (point Selected in search, target point to navigate or next turn point in active navigation) changes sector. 
- Smart autoannounance off - you will see the toast every 10 s.

**Comment by Zirochkabila** [comment]
# Testing 1
`OsmAnd~ 5.0.0#4273m, Released: 2025-01-31`

Setting the interval in the Autoannounce period does not affect Smart Autoannounce in any way - deviations from the track are announced in the same way as in normal navigation

**Comment by Zirochkabila** [comment]
# Testing 2
`OsmAnd~ 5.0.0#4312m, Released: 2025-02-05`

When Accessibility Mode is activated, the built-in OsmAnd functions display notifications with the specified frequency (Enable smart auto announce and set the period to 10s, for example), but only on the screen in the form of text notifications (this is not quite suitable for visually impaired people) and only after this function has been activated by clicking on My Location and activating it
In order for notifications to be voiced with the required frequency, you need to additionally enable TalkBack in the system settings of the device itself.
I think this should be emphasized in the documentation @shurasmiling [Docs](https://osmand.net/docs/user/plugins/accessibility#plugin-settings)




<details><summary>Video example</summary>
<table style="width:100%">
  <tr>
    <th>TalkBack on</th>
    <th>TalkBack off</th>
  </tr>
  <tr>
    <td><video source src="https://github.com/user-attachments/assets/e9ae3968-f6bf-4b23-ab1c-8953a336d01f"  width="300" height="auto"/></td>
    <td><video source src="https://github.com/user-attachments/assets/f4de5217-3f61-4938-a5b0-9aebe302f78e"  width="300" height="auto"/> </td>
  </tr>
</table> 
</details>

**Comment by vshcherb** [comment]
> in the form of text notifications (this is not quite suitable for visually impaired people)

This is implemented intentionally cause speak over should be enabled

**Comment by shurasmiling** [comment]
https://osmand.net/docs/user/plugins/accessibility

**Comment by vshcherb** [comment]
Accessibility documentation doesn't mention long tap on My location to enable/edisable menu Show you position/ Show your details and Start/Stop autoannouncements - see video https://github.com/osmandapp/OsmAnd/issues/19209#issuecomment-2636871301

**Comment by vshcherb** [comment]
Documentation:
- How to enable announcements and how do they look?
- Do we need navigation or point for announcements?
- Do we need to enable periodic announcements to have smart autonannce ?
- Does smart only works when user deviates from route or also works periodically? What's the threshold?
